### ISSUE TITLE
'solvespace-cli regenerate' does not update file

### ISSUE THREAD

**Issue by justthinkering** [solvespace/solvespace#323]
### System information

SolveSpace version: 3.0~16540b1b

Operating system: Linux 4.9.83, NixOS, 17.09.3047.8bce347f02f (Hummingbird)

### Expected behavior

The command ```solvespace-cli regenerate test.slvs``` (with test.slvs being an arbitrary SolveSpace file) should update and rewrite the file 'test.slvs'.

### Actual behavior

After some warnings, only '.slvs' is written, but 'test.slvs' remains the same.

```
Missing (absent) translation for group-name'#references'
Missing (absent) translation for group-name'sketch-in-plane'
Written '.slvs'.
```

**Comment by whitequark** [comment]
Hm, this is odd. Does NixOS apply any patches? Can you please post the log file produced by running `strace solvespace-cli regenerate test.slvs`?

**Comment by justthinkering** [comment]
As far as I understand NixOS, the SolveSpace package is defined in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/graphics/solvespace/default.nix.

Please find the strace output in [logfile.txt](https://github.com/solvespace/solvespace/files/1984918/logfile.txt).

**Comment by whitequark** [comment]
The NixOS package lists the version 2.3, which does not have the solvespace-cli tool.

**Comment by justthinkering** [comment]
I guess, they forgot to update the name "solvespace-2.3-20170808", but they link to the SolveSpace revision "16540b1b2c540a4b44500ac02aaa4493bccfba7e" (https://github.com/solvespace/solvespace/commit/16540b1b2c540a4b44500ac02aaa4493bccfba7e), which agrees with my version "3.0~16540b1b" (from Help → About...). I followed the default way to install the NixOS package and did not customize anything.

**Comment by justthinkering** [comment]
The patched command
```
./solvespace-cli regenerate test.slvs
```
results in
```
outputFile1: %.slvs
parent: , name: test.slvs, stem: test
outputSubst1: 
outputSubst2: 
outputFile2: .slvs
Missing (absent) translation for group-name'#references'
Missing (absent) translation for group-name'sketch-in-plane'
Generate::REGEN took 169 ms
Written '.slvs'.
```
and only writes the '.slvs', but adding a "path" before the filename
```
./solvespace-cli regenerate ./test.slvs 
```
gives
```
outputFile1: %.slvs
parent: ./, name: test.slvs, stem: test
outputSubst1: ./
outputSubst2: ./test
outputFile2: ./test.slvs
Missing (absent) translation for group-name'#references'
Missing (absent) translation for group-name'sketch-in-plane'
Written './test.slvs'.
```
and updates the file 'test.slvs'.

**Comment by whitequark** [comment]
Fixed in 2b9ffd1. Thanks for the report.
